pilot/pkg/networking/core/v1alpha3/httproute_test.go

```diff
@@ -417,7 +478,7 @@ func TestSidecarOutboundHTTPRouteConfig(t *testing.T) {
 			sidecarConfig:         sidecarConfig,
 			virtualServiceConfigs: nil,
 			expectedHosts: map[string]map[string]bool{
-				"test.com:8080": {"test.com:8080": true, "8.8.8.8:8080": true},
+				"test.com:8080": {"test.com": true, "test.com:8080": true, "8.8.8.8": true, "8.8.8.8:8080": true},
```

Comment:
Why this change?

Reply:
That is because earlier with out this change, this route is treated as http proxy route and we used to set listenerPort  = 0 and if it is zero, we merge https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/httproute.go#L346 here, which merged all the virtual hosts, Now with this change since it is not http_proxy, we do not merge - so we get all virtual hosts.

Reply:
IC, please update the test comments as well

Reply:
Sorry, Did not notice the test comments. I have updated now. PTAL